frontend/react/src/components/layout/DateRange.js

```diff
@@ -1,296 +1,332 @@
-            <div className="errors">
-              {this.state.endRangeErr === true ? (
-                <div> End date must come after start date</div>
-              ) : null}
-            </div>
-          </Fragment>
-        </div>
-      </Fragment>
-    );
-  }
-}
-
-DateRange.propTypes = {
-  previousEntry: PropTypes.bool,
-};
-
-const mapStateToProps = (state) => ({
-  year: state.stateUser.formYear,
-});
-
-export default connect(mapStateToProps)(DateRange);
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { TextField } from "@cmsgov/design-system-core";
+import PropTypes from "prop-types";
+import uuid from "react-uuid";
+
+// This method checks that month input is appropriate:
+// (not empty, max of 2 digits, no letters, between 1 & 12)
+const validateMonth = (input) => {
+  let returnString;
+
+  // Handles an empty input field
+  if (input === "") {
+    returnString = "Month field cannot be empty";
+  }
+
+  // Prevents users from putting in more than 2 characters
+  if (input.length > 2) {
+    returnString = "Month length must not exceed 2";
+  }
+
+  // Checks for non-numeric characters
+  if (Number.isNaN(parseInt(input, 10)) || /^\d+$/.test(input) === false) {
+    returnString = "Please enter a number";
+  }
+
+  if (parseInt(input, 10) < 1 || parseInt(input, 10) > 12) {
+    // Checks that the month value is within a normal range
+    returnString = "Please enter a valid month number";
+  }
+  return returnString;
+};
+
+class DateRange extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      endRangeErr: false,
+      monthStart: "",
+      yearStart: "",
+      monthEnd: "",
+      yearEnd: "",
+      startErrorMessage: [],
+      endErrorMessage: [],
+    };
+    this.handleInput = this.handleInput.bind(this);
+
+    this.checkChronology = this.checkChronology.bind(this);
+
+    this.validateStartInput = this.validateStartInput.bind(this);
+    this.validateEndInput = this.validateEndInput.bind(this);
+
+    this.validateYear = this.validateYear.bind(this);
+  }
+
+  componentDidMount() {
+    // Stored value example: ['2019-11-01', '2020-09-01']
+    const { question } = this.props;
+    const storedValue = question.answer.entry;
+
+    if (storedValue) {
+      // Returns an array with the values of the start date, split
+      const startDate = storedValue[0].split("-"); // ie: ['2019', '11', '01']
+
+      // Returns an array with the values of the start date, split
+      const endDate = storedValue[1].split("-"); // ie: ['2020', '09', '01']
+
+      const monthStartValue = startDate[1]; // '11'
+      const yearStartValue = startDate[0]; // '2019'
+
+      const monthEndValue = endDate[1]; // '09'
+      const yearEndValue = endDate[0]; // '2020'
+
+      this.setState({
+        monthStart: monthStartValue || "",
+        monthEnd: monthEndValue || "",
+        yearStart: yearStartValue || "",
+        yearEnd: yearEndValue || "",
```

Comment:
Someone on the CL team started using null coalescing elsewhere, so it'd make sense to either use it everywhere, or take it out of those other places. I think it's cleaner and implies the intention better, but I don't feel very strongly about it, to be honest. Just flagging it for the team to think about. 😄 

```suggestion
        monthStart: monthStartValue ?? "",
        monthEnd: monthEndValue ?? "",
        yearStart: yearStartValue ?? "",
        yearEnd: yearEndValue ?? "",
```